plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java

```diff
@@ -1760,6 +1760,40 @@ private TableStatistics getTableStatistics(String tableName, Constraint constrai
                 });
     }
 
+    @Test
+    public void testPartitionValuesWithSpecialCharacters()
+    {
+        assertUpdate("CREATE TABLE iceberg.tpch.test_special_characters (col_1 INTEGER, col_2 VARCHAR) WITH (partitioning = ARRAY['col_2'])");
+        String values = "" +
+                "(1, 'abcd ef'), " +
+                "(2, 'zip per'), " +
+                "(3, 'race car'), " +
+                "(4, 'with:colon'), " +
+                "(5, 'with-hyphen'), " +
+                "(6, 'with?question'), " +
+                "(7, ' startof'), " +
+                "(8, 'endof '), " +
+                "(9, 'with👨‍🏭factoryman'), " +
```

Comment:
the fact this happens to be "factory man" is probably not important

i guess the point here is "non-BMP with combining characters" or "non-BMP with non-BMP combining characters", right?